### PR TITLE
[SPARK-31038][SQL] Add checkValue for spark.sql.session.timeZone

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

The `spark.sql.session.timeZone` config can accept any string value including invalid time zone ids, then it will fail other queries that rely on the time zone. We should do the value checking in the set phase and fail fast if the zone value is invalid.

### Why are the changes needed?

improve configuration
### Does this PR introduce any user-facing change?

yes, will fail fast if the value is a wrong timezone id
### How was this patch tested?

add ut

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @maropu, thanks for review.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -1654,6 +1656,8 @@ object SQLConf {
       .doc("""The ID of session local timezone, e.g. "GMT", "America/Los_Angeles", etc.""")
       .version("")
       .stringConf
+      .checkValue(z => Try { DateTimeUtils.getZoneId(z) }.isSuccess, "The timeZone should obey" +
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: can we create a method in `object SQLConf`?
```
private def isValidTimezone...
```

File: sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala [hunk 2]
```diff
@@ -348,4 +348,24 @@ class SQLConfSuite extends QueryTest with SharedSparkSession {
...
+      spark.conf.set(SQLConf.SESSION_LOCAL_TIMEZONE.key, "pst")
+    }
+    intercept[IllegalArgumentException] {
+      spark.conf.set(SQLConf.SESSION_LOCAL_TIMEZONE.key, "GMT+8:00")
+    }
+    intercept[IllegalArgumentException] {
+      spark.conf.set(SQLConf.SESSION_LOCAL_TIMEZONE.key, "Asia/shanghai")
+    }
+
+
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala]: remove extra blank lines

**Review by cloud-fan — APPROVED** [review]
good catch!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 3]
```diff
@@ -75,6 +77,10 @@ object SQLConf {
     }
   }
 
+  def isValidTimezone(zone: String): Boolean = {
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: can we move it closer to where it's used?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 4]
```diff
@@ -1654,6 +1660,8 @@ object SQLConf {
       .doc("""The ID of session local timezone, e.g. "GMT", "America/Los_Angeles", etc.""")
       .version("")
       .stringConf
+      .checkValue(isValidTimezone, "The time zone should obey rules managed by" +
+        " java.time.zone.ZoneRulesProvider")
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: This is still unclear, maybe just say: `Cannot resolve the give timezone with ZoneId.of(_, ZoneId.SHORT_IDS)`

File: sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala [hunk 5]
```diff
@@ -348,4 +348,22 @@ class SQLConfSuite extends QueryTest with SharedSparkSession {
...
+      "America/Chicago")
+
+    intercept[IllegalArgumentException] {
+      spark.conf.set(SQLConf.SESSION_LOCAL_TIMEZONE.key, "pst")
+    }
+    intercept[IllegalArgumentException] {
+      spark.conf.set(SQLConf.SESSION_LOCAL_TIMEZONE.key, "GMT+8:00")
+    }
+    intercept[IllegalArgumentException] {
+      spark.conf.set(SQLConf.SESSION_LOCAL_TIMEZONE.key, "Asia/shanghai")
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala]: Please check the error messages here?

**Review by maropu — APPROVED** [review]
LGTM except for the existing @cloud-fan comments.

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Pull request closed without merging**